### DISCUSSION TITLE
Shorting stocks

### DISCUSSION THREAD

**Discussion by mostlyvirtual** [ghostfolio/ghostfolio#1677] [Ideas]
I used the search feature, but couldn't find anything about how you handle stock shorting. i.e. selling 1 share of AAPL if you own zero shares of AAPL.

The logic would be that selling 1 share of AAPL at $100 (aka shorting) would add the share with a -1 to the portfolio, which I see it happens, but the $100 should go to cash balance, which doesn't seem to happen.

**Comment by dtslvr** [comment]
Hi @mostlyvirtual

The cash balance is not adjusted automatically, not even in a conventional sell. I think it would not be consistent if it happened for shorting. What do you think?

**Reply by mostlyvirtual** [reply]
Thank you for the reply. I didn't realize the cash balance is not adjusted automatically at this stage. In that case, it makes sense that you also don't adjust it for shorting.
However, the portfolio value might still need a tweak. i.e. if I have a portfolio valued at $1000 and I short a stock worth $100, now my portfolio value is $900, whereas it should still be $1000 and then increase if the stock goes down, or decrease if the shorted stock goes up.

An idea might be to treat it exactly as a buy position, but track it inversely if it exists in the portfolio, as a holding - because if it exists as a holding with a minus, then it must be a short position. That would mean the import feature would still work.
i.e. if I buy one share of AAPL and it goes up by $10, then the portfolio goes up - the inverse of that would be if I short 1 share of AAPL and it goes up by $10, then the portfolio goes down by $10

**Reply by dtslvr** [reply]
Okay, the starting point would be to model a minimal scenario as a test case. Here is an example: [portfolio-calculator-novn-buy-and-sell.spec.ts](https://github.com/ghostfolio/ghostfolio/blob/main/apps/api/src/app/portfolio/portfolio-calculator-novn-buy-and-sell.spec.ts). You can find more examples [here](https://github.com/ghostfolio/ghostfolio/tree/main/apps/api/src/app/portfolio) (`*.spec.ts`).

The tests can be executed via `yarn test`.
